**Title: TBD **

Yuka Takemon, ^\*^ Daniel M. Gatti, ^\*^ Susan M. Sheehan, ^\*^ Holly S.
Savage, ^\*^ Ron Korstanje ^\*^

^\*^ The Jackson Laboratory, Bar Harbor, ME, USA.

Alport Syndrome (AS) is a hereditary disease affecting 1 in 5,000 births
in the United States^1^. Children and young adults with AS suffer from
hearing loss, vision abnormalities, and most notably kidney
disease^2,3^. AS patients with kidney disease are diagnosed with
glomerulonephritis and present with symptoms of hematuria and
proteinuria^4,5^. Inevitably, their kidney function progressively
deteriorates and leads to End-Stage Renal Failure (ESRF) ^3,6,7^. It is
well studied that ESRF in AS is caused by genetic mutations in the α3,
α4, and α5 chains of type IV collagen, which are encoded by *COL4A3,
COL4A4,* and *COL4A5*^2,4,8-10^. In the kidney, the three type IV
collagen proteins form heterotrimers and are exclusively found in the
glomerular basement membrane (GBM) ^8,10,11^. A dysfunction in any one
of the type IV collagen proteins causes the GBM to weaken and distend,
and podocytes foot process effacement to occur^7^. COL4A5 is the only
type IV collagen protein encoded on the X chromosome, and is responsible
for 80% of AS diagnoses^5^. As the X-linkage suggests, males that are
hemizygous to the *COL4A5* mutation are disproportionally affected
compared to females. Males have an earlier onset and increased severity
of the disease, with 50% of patients requiring dialysis or kidney
transplants due to ESRF by the age of 25 and 100% by the age of 60^12^.
On the other hand females that are heterozygous for this X-linked AS
have relatively later onset with 12% developing ESRF by the age of 40
and 40% at the age of 80 years^5,13^.

Patients with similar genetic mutations do not all present the disease
in a similar manner, and studies have observed their age of onset and
severity to be highly variable^12^. Clinical outcomes of patients with
similar familial mutations that present with hematuria are variable,
ranging from normal renal function throughout life to chronic kidney
disease to ESRF^10,14^. It is widely accepted that the variable clinical
outcomes such as the age of onset and severity of AS is in part due to
underlying genetic mechanisms that are able to modify disease
progression^5,15-17^. Such genetic mechanisms are hereinafter termed
modifier genes. Modifier genes act as independent genes that have the
ability to affect the expressivity of phenotypes to either extremes^18^,
and in the case of AS, the age of onset and severity. Studies to map
modifier genes in humans AS patient cohorts have been attempted, however
due to variable environmental factors, insufficient sample size, and
other confounding factors make it difficult to draw confident
conclusions^19-21^. For these reasons mammalian AS models such as mice
would be more appropriate as their external environment, genetic
background, and sample size can be controlled to meet the needs of an
experimental design.

To our knowledge there has only been one study conducted to identify
modifier genes in AS models^16^. Andrews K. L. et al. observed
variations in phenotype between *Col4a3* knock out mice in a 129X1/SvJ,
C57BL/6J, F1 cross between 129X1/SvJ and C57BL/6J, and a F1 to C57BL/6J
back cross mouse models, which highlighted the presence of modifier
genes (explain more).

and identified 2 quantitative trail loci (QTL) on chromosome 9 and 16,
however the intervals were not narrow enough to confidently identify
candidate genes^16^. Since 2002, when the prior mentioned study was
conduction, there have been major technological advancements and
recourses to allow for high-resolution mapping. The Diversity Outbred
(DO) mouse population, first published in 2012, is a genetically
heterogeneous model derived from multi-parent crosses consisting of 5
classical inbred models (A/J, C57BL/6J, 129S1/SvImJ, NOD/ShiLtJ, and
NZO/HlLtJ) and 3 wild-derived models (CAST/EiJ, PWK/PhJ, and
WSB/EiJ)^22^. Each individual mouse in a DO population is a genetically
unique combination of the 8 founder strains, and best reflects the
diversity seen in human populations^22,23^. Furthermore, the development
of the third generation of the Mouse Universal Genome Array series,
GigaMUGA, allow for high-resolution mapping at 143,259 SNPs, and the
ability to detect parental origin haplotypes in a DO population^24^.
Utilizing these aforementioned recourses with enough sample size would
be ideal for mapping modifier genes in AS.

In this study we aimed to effectively identify modifier genes in
X-linked AS by introducing the *Col4a5* mutation into a diverse
geneti­­­c background using the DO mouse model. The founder strains that
make up DO mice contribute 90% of known genetic variations found in
laboratory mice, and the captured genetic variations are randomly
distributed across the genome^25,26^. Each F1 mouse with an X-linked AS
mutation influenced by a unique combination of heterogeneous genetic
background will present a range of renal phenotypes representative of
the human AS population. With a sufficient samples size through this
method, we will have the ability for high-resolution mapping, and
identify modifier genes in X-linked AS. The identification of modifier
genes will allow for the design of targeted therapeutics, an
intervention urgently needed in for AS patients around the world.

1\. Hertz JM, Thomassen M, Storey H, Flinter F: Clinical utility gene
card for: Alport syndrome. *European Journal of Human Genetics* 20: 2012

2\. Kruegel J, Rubel D, Gross O: Alport syndrome--insights from basic and
clinical research. *Nat Rev Nephrol* 9: 170–178, 2013

3\. Savige J, Gregory M, Gross O, Kashtan C, Ding J, Flinter F: Expert
guidelines for the management of Alport syndrome and thin basement
membrane nephropathy. *J. Am. Soc. Nephrol.* 24: 364–375, 2013

4\. Kashtan CE: Alport syndrome and thin glomerular basement membrane
disease. *Journal of the American Society of Nephrology* 9: 1736–1750,
1998

5\. Rheault MN: Women and Alport syndrome. *Pediatr Nephrol* 27: 41–46,
2012

6\. Kelly YP, Patil A, Wallis L, Murray S, Kant S, Kaballo MA, Casserly
L, Doyle B, Dorman A, O'Kelly P, Conlon PJ: Outcomes of kidney
transplantation in Alport syndrome compared with other forms of renal
disease. *Ren Fail* 1–4, 2016

7\. Cosgrove D, Liu S: Collagen IV diseases: A focus on the glomerular
basement membrane in Alport syndrome. *Matrix Biology* 1–10, 2016

8\. Hudson BG, Hudson BG, Tryggvason K, Tryggvason K, Sundaramoorthy M,
Neilson EG: Alport“s syndrome, Goodpasture”s syndrome, and type IV
collagen. *N. Engl. J. Med.* 348: 2543–2556, 2003

9\. Savige J: Alport syndrome: its effects on the glomerular filtration
barrier and implications for future treatment. *J Physiol* 592:
4013–4023, 2014

10\. Kashtan C: Alport syndrome: facts and opinions. *F1000Res* 6: 50,
2017

11\. Heidet L, Cai Y, Guicharnaud L, Antignac C, Gubler MC: Glomerular
expression of type IV collagen chains in normal and X-linked Alport
syndrome kidneys. *Am. J. Pathol.* 156: 1901–1910, 2000

12\. Kashtan CE, Ding J, Gregory M, Gross O, Heidet L, Knebelmann B,
Rheault M, Licht C: Clinical practice recommendations for the treatment
of Alport syndrome: a statement of the Alport Syndrome Research
Collaborative. *Pediatr Nephrol* 28: 5–11, 2013

13\. Jais JP, Knebelmann B, Giatras I, De Marchi M, Rizzoni G, Renieri A,
Weber M, Gross O, Netzer K-O, Flinter F, Pirson Y, Dahan K, Wieslander
J, Persson U, Tryggvason K, Martin P, Hertz JM, Schröder C, Sanak M,
Carvalho MF, Saus J, Antignac C, Smeets H, Gubler MC: X-linked Alport
syndrome: natural history and genotype-phenotype correlations in girls
and women belonging to 195 families: a “European Community Alport
Syndrome Concerted Action” study. *Journal of the American Society of
Nephrology* 14: 2603–2610, 2003

14\. Longo I, Porcedda P, Mari F, Giachino D, Meloni I, Deplano C, Brusco
A, Bosio M, Massella L, Lavoratti G, Roccatello D, Frascá G, Mazzucco G,
Muda AO, Conti M, Fasciolo F, Arrondel C, Heidet L, Renieri A, De Marchi
M: COL4A3/COL4A4 mutations: From familial hematuria to
autosomal-dominant or recessive Alport syndrome. *Kidney International*
61: 1–10, 2010

15\. Flinter F, Savige J, Savva I, Renieri A, Knebelman B, Ding J, Zhang
Y, Gale DP, Schachter AD, Morton LCG, Huang M, Schifter J, Skelding J,
Friede T, Lennon R: Advances and unmet needs in genetic, basic and
clinical science in Alport syndrome: report from the 2015 International
Workshop on Alport Syndrome. *Nephrol. Dial. Transplant.* gfw095, 2016

16\. Andrews KL, Mudd JL, Li C, Miner JH: Quantitative trait loci
influence renal disease progression in a mouse model of Alport syndrome.
*Am. J. Pathol.* 160: 721–730, 2002

17\. Korstanje R, Caputo CR, Doty RA, Cook SA, Bronson RT, Davisson MT,
Miner JH: A mouse Col4a4 mutation causing Alport glomerulosclerosis with
abnormal collagen α3α4α5(IV) trimers. *Kidney International* 85:
1461–1468, 2014

18\. Nadeau JH: Modifier genes in mice and humans. *Nature Reviews
Genetics* 2: 165–174, 2001

19\. Kamiyoshi N, Nozu K, Fu XJ, Morisada N, Nozu Y, Ye MJ, Imafuku A,
Miura K, Yamamura T, Minamikawa S, Shono A, Ninchoji T, Morioka I,
Nakanishi K, Yoshikawa N, Kaito H, Iijima K: Genetic, Clinical, and
Pathologic Backgrounds of Patients with Autosomal Dominant Alport
Syndrome. *Clin J Am Soc Nephrol* 11: 1441–1449, 2016

20\. Deltas C, Pierides A, Voskarides K: Molecular genetics of familial
hematuric diseases. *Nephrol. Dial. Transplant.* 28: 2946–2960, 2013

21\. Papazachariou L, Demosthenous P, Pieri M, Papagregoriou G, Savva I,
Stavrou C, Zavros M, Athanasiou Y, Ioannou K, Patsias C, Panagides A,
Potamitis C, Demetriou K, Prikis M, Hadjigavriel M, Kkolou M, Loukaidou
P, Pastelli A, Michael A, Lazarou A, Arsali M, Damianou L, Goutziamani
I, Soloukides A, Yioukas L, Elia A, Zouvani I, Polycarpou P, Pierides A,
Voskarides K, Deltas C: Frequency of COL4A3/COL4A4 mutations amongst
families segregating glomerular microscopic hematuria and evidence for
activation of the unfolded protein response. Focal and segmental
glomerulosclerosis is a frequent development during ageing. *PLoS ONE*
9: e115015–, 2014

22\. Churchill GA, Gatti DM, Munger SC, Svenson KL: The Diversity Outbred
mouse population. *Mamm Genome* 23: 713–718, 2012

23\. Bogue MA, Churchill GA, Chesler EJ: Collaborative Cross and
Diversity Outbred data resources in the Mouse Phenome Database. *Mamm
Genome* 26: 511–520, 2015

24\. Morgan AP, Fu C-P, Kao C-Y, Welsh CE, Didion JP, Yadgary L, Hyacinth
L, Ferris MT, Bell TA, Miller DR, Giusti-Rodriguez P, Nonneman RJ, Cook
KD, Whitmire JK, Gralinski LE, Keller M, Attie AD, Churchill GA, Petkov
P, Sullivan PF, Brennan JR, McMillan L, Pardo-Manuel de Villena F: The
Mouse Universal Genotyping Array: From Substrains to Subspecies. *G3
(Bethesda)* 6: 263–279, 2015

25\. Roberts A, Pardo-Manuel de Villena F, Wang W, McMillan L, Threadgill
DW: The polymorphism architecture of mouse genetic resources elucidated
using genome-wide resequencing data: implications for QTL discovery and
systems genetics. *Mamm Genome* 18: 473–481, 2007

26\. Threadgill DW, Churchill GA: Ten years of the collaborative cross.
*G3 (Bethesda)* 2: 153–156, 2012
