**COMPLETE METHODS**

**Animals**

Col4a5 knockout mice, B6.Cg-*Col4a5^tm1Yseq^*/J (Stock\#006183), was
developed by Dr. Yoav Segal using a targeting vector containing a loxP
site flanked neomycin resistance gene and a G213T point mutation was
introduced into exon. The construct was electroporated into 129SvJ
derived ESVJ-1182 embryonic stem (ES) cells. ES cells containing the
point mutation were injected into C57BL/6J (Stock\#000664). The neo
cassettes were removed by crossing to FVB/N-Tg(ACTB-cre) 2Mrt/J
(Stock\#3376) mice to and then backcrossed to C57BL/6J for 15
generations. As a result regions immediately surrounding the point
mutation contain residual 129SvJ markers.

Col4a5 knockout mice for this experiment were rederived from
cryopreservation at The Jackson Laboratory, with the females maintained
as heterozygous for the Col4a5 mutation and hemizygous in males. 100
heterozygous female Col4a5 mutant mice were crossed with 100 unique
diversity outbred males, J:DO (JR\#009376), to generate a cohort of 100
males and 100 female F1 animals, where females were heterozygous of the
Col4a5 mutation and males were homozygous. Each of the 200 F1 animals
carry a single copy of C57BL/6J at each chromosome, while the other copy
is contributed form on DO background. The DO background is a unique
mixture of eight founder strains, which include five classical inbred
strains (129S1/SvImJ, A/J, C57BL/6J, NOD/ShiLtJ, NZO/HILtJ) and three
wild-derived strains (CAST/EiJ, PWK/PhJ and WSB/EiJ). Thus at a given
locus each F1 animal will have a C57BL6/J allele, and one of eight
founder strain allele.

**Genotyping and quality control**

All 200 mice were fully genotyped for 143,259 SNPs by Neogen Genomics
using the Giga Mouse Universal Genotyping Array (GigaMUGA) built on an
Illumina Infinium platform.

(need Dan’s input here about genome probability construction)

Only 182 samples passed quality control (QC) from genome probability
reconstruction.

Simultaneously we reconstructed the genome for the 192 mice we had RNA
sequencing (RNA-seq) data for using the Expectation-Maximization
algorithm for Allele Specific Expression (EMASE) software developed by
the Churchill group (ref). EMASE algorithm takes diploid transcript tom
alignment and estimates the expression abundance for each allele. The
seemingly redundant effort was essential for our rigorous quality
control for sample switching, deviation in allele frequency, and sample
recover of the GigaMUGA data. Our comparison between the 182 mice with
both GigaMUGA and RNA-seq data confirmed correct genotyping for 157
GigaMUGA and 24 samples that could not be confirmed. Upon further
investigation revealed 11 of the 24 unconfirmed samples to be swapped
with another Neogen Genomic customer. We were able to replace some
missing samples using RNA-seq reconstruction, however the 8 samples that
did not have RNA-seq data were removed from the study, as we could not
verify their integrity. Through these stringent processes, we were able
to confirm 192 quality samples for our analysis.

GFR from Far2 paper

MME and glomerular filtration rate (GFR) were compared between knockout
and wildtype mice at 6, 12, and 18 months of age (Figure 1C). At 6
months of age both groups had a low MME score (wildtype: 66%, knockout:
72%) and there was no significant difference between the two groups.
However, at 12 months the MME score in the wildtype animals increased
significantly, while this did not happen in the knockout animals
(wildtype: 89%, knockout 68%, P=1.27x10^-6^). At 18 months, the knockout
animals showed the same high amount (89%) of MME as the wildtype animals
(88%). A significant difference (P=0.0118) was observed for GFR is at 6
months, with higher GFR (481±64 μl/min) in the knockout animals than the
wildtype animals (381±79 μl/min) (Figure 1D). Therefore, our data shows
that deletion of *Far2* leads to a delay in MME and an improvement of
renal function at a young age.
