**RESULTS**

**Figure 1.** Col4a5 mutant model, B6.Cg-*Col4a5^tm1Yseq^*/J, mating
scheme with diversity outbred mice, J:DO. (A) 100 heterozygous female
Col4a5 models, B6.Cg-*Col4a5^tm1Yseq^*/J, were crossed with 100
diversity outbred mice, J:DO. From each mating pairs we used one F1
female that was heterozygous to the Col4a5 mutation, and one F1 male
that was hemizygous to the Col4a5 mutaiton for a total of 200 mice in
the experimental cohort. Due to X-inactivation, F1 females will have a
mosaic of Col4a5 knock out and wild-type cells. (B) Col4a5 mutation will
contain residual 129X1/SvJ locus directly surrounding mutation site, due
ES strain origin of the original knockout model.

**Figure 2.** Phenotyping of F1 females and males for renal function.
100 animals of each sex were phenotyped for (left panel) glomerular
filtration rate (GFR) at 14 weeks of age, (middle panel) albumin to
creatinine ratio (ACR) at 6, 10, and 15 weeks of age, and (right panel)
correlation between GFR at 14 weeks and ACR at 15 weeks of age. (A1)
Distribution of females at 14 weeks according to their GFR display
higher functioning kidneys (n =, average GFR =, stderror = ) compared to
the distribution of males (B1, n =, average GFR =, stderror = ). (A2)
Distribution of females at 6, 10, and 15 weeks displayed in red, green,
and blue respectively show increasing ACR with age (significant
difference between ages?). (B2) Comparatively males also show increasing
ACR with age (significant ACR with age), furthermore, ACR values in
males show higher ACR values than females at their corresponding time
points suggesting increase severity of renal disease in males. (A3)
Correlation between ACR at 15 weeks and GFR at 14 weeks of age in
females show very slight negative correlation (R2 = 0.048, p-val = ).
(B3) Correlation between ACR at 15 weeks and GFR at 14 weeks of age in
males show negative correlation (R2 = 0.411, p-val = ). Overall males
show severe renal phenotype compared to their female counterparts.

**Figure 3.** Kinship maps showing heterogeneity of F1 experimental
cohort before and after quality control (QC). (A1) Kinship map of
samples pre-QC show blocks of samples that have nearly identical genomic
back ground, suggesting either sample duplications or introduction of
non-DO animals in cohort. (A2) Kinship map of samples post-QC show
complete heterogeneity of data set. Previously identified incorrect
GigaMUGA data were replaced with RNA reconstructed genome data and
recovered GigaMUGA data.

**Figure 4. 1** GFR qtl

**Figure 4.** Haplotype QTL

**Figure 4.2** Albumin qtl 6wks

**Figure 4.3** Albumin qtl 10wks

**Figure 4.4** Albumin qtl 15wks

**Figure 5** Xce allele
